src/wh_comm.c

```diff
@@ -108,6 +108,11 @@ int wh_CommClient_SendRequest(whCommClient* context, uint16_t magic,
         return WH_ERROR_BADARGS;
     }
 
+    /* Check if the data size is within allowed limits */
+    if (data_size > WOLFHSM_CFG_COMM_DATA_LEN) {
```

Comment:
How did we miss this??  I assume the lower-level does a sanity check but we should have caught it here.  Nice!